backend/src/main/java/ch/sectioninformatique/template/user/UserService.java

```diff
@@ -0,0 +1,50 @@
+package ch.sectioninformatique.template.user;
+
+import ch.sectioninformatique.template.auth.RegisterUserDto;
+import ch.sectioninformatique.template.security.Role;
+import ch.sectioninformatique.template.security.RoleEnum;
+import ch.sectioninformatique.template.security.RoleRepository;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+@Service
+public class UserService {
+    private final UserRepository userRepository;
+    private final RoleRepository roleRepository;
+    private final PasswordEncoder passwordEncoder;
+
+
+    public UserService(UserRepository userRepository,
+                       RoleRepository roleRepository,
+                       PasswordEncoder passwordEncoder)
+    {
+        this.userRepository = userRepository;
+        this.roleRepository = roleRepository;
+        this.passwordEncoder = passwordEncoder;
+    }
+
+    public List<User> allUsers() {
+        List<User> users = new ArrayList<>();
+
+        userRepository.findAll().forEach(users::add);
+
+        return users;
+    }
+    
+    public User createAdministrator(RegisterUserDto input) {
```

Comment:
Instead of "createAdministrator" method, shoudn't we have a "setAdminRole" or "giveAdminRole" method to give admin authorizations to an existing user ?